luigi/worker.py

```diff
@@ -117,14 +117,19 @@ def run(self):
 
         status = FAILED
         error_message = ''
-        missing = []
+        missing_taskids = []
+        missing_targets = []
         new_deps = []
         try:
             # Verify that all the tasks are fulfilled!
-            missing = [dep.task_id for dep in self.task.deps() if not dep.complete()]
-            if missing:
-                deps = 'dependency' if len(missing) == 1 else 'dependencies'
-                raise RuntimeError('Unfulfilled %s at run time: %s' % (deps, ', '.join(missing)))
+            missing_deps = [dep for dep in self.task.deps() if not dep.complete()]
+            missing_targets = [tgt for tgt in flatten(dep.output()) for dep in missing_deps if not tgt.exists()]
```

Comment:
Can you assume that there is a `output()` defined on all tasks? I don't think so, it's just a common practice but in general the interface is only the run-requires-complete triple iirc.